sys/net/iflib.c

```diff
@@ -6428,33 +6428,29 @@ iflib_legacy_setup(if_ctx_t ctx, driver_filter_t filter, void *filter_arg, int *
 	device_t dev;
 	struct grouptask *gtask;
 	struct resource *res;
-	struct taskqgroup *tqg;
-	void *q;
 	int err, tqrid;
 	bool rx_only;
 
-	q = &ctx->ifc_rxqs[0];
-	info = &rxq[0].ifr_filter_info;
-	gtask = &rxq[0].ifr_task;
-	tqg = qgroup_if_io_tqg;
+	info = &rxq->ifr_filter_info;
+	gtask = &rxq->ifr_task;
 	tqrid = *rid;
 	rx_only = (ctx->ifc_sctx->isc_flags & IFLIB_SINGLE_IRQ_RX_ONLY) != 0;
 
 	ctx->ifc_flags |= IFC_LEGACY;
 	info->ifi_filter = filter;
 	info->ifi_filter_arg = filter_arg;
 	info->ifi_task = gtask;
-	info->ifi_ctx = rx_only ? ctx : q;
+	info->ifi_ctx = rxq;
 
 	dev = ctx->ifc_dev;
 	/* We allocate a single interrupt resource */
-	err = _iflib_irq_alloc(ctx, irq, tqrid, rx_only ? iflib_fast_intr_ctx :
+	err = _iflib_irq_alloc(ctx, irq, tqrid, rx_only ? iflib_fast_intr :
```

Comment:
I don't really understand why we have both `iflib_fast_intr` and `iflib_fast_intr_ctx` TBH.  The only difference is the `ctx` variant checks if the gtask taskqueue is NULL.

Reply:
iflib_fast_intr_ctx was added in #95246abb21da along with iflib_fast_intr_rxtx. I my understanding the intention was to have separate handler for admin/link interrupts, though at that point there was no actual difference between iflib_fast_intr and iflib_fast_intr_ctx. With this patch iflib_fast_intr_ctx starts using taskqueue while iflib_fast_intr keeps using taskqgroup.